packages/components/data-table-manager/src/column-settings-manager/column-settings-manager.tsx

```diff
@@ -19,20 +26,47 @@ import DroppablePanel from '../droppable-panel';
 import SettingsContainer from '../settings-container';
 import messages from './messages';
 import { HIDDEN_COLUMNS_PANEL, SELECTED_COLUMNS_PANEL } from './constants';
+import { warning } from '@commercetools-uikit/utils';
+
+type TColumnData = {
+  key: string;
+  label: ReactNode;
+};
+
+type TColumnSettingsManagerProps = {
+  availableColumns: TColumnData[];
+  selectedColumns: TColumnData[];
+  onUpdateColumns: (updatedColums: TColumnData[]) => void;
+  areHiddenColumnsSearchable?: boolean;
+  searchHiddenColumns?: (searchTerm: string) => Promise<unknown>;
+  searchHiddenColumnsPlaceholder?: string;
+
+  onClose: (
+    event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
+  ) => void;
+  primaryButton?: ReactElement;
+  secondaryButton?: ReactElement;
+  managerTheme?: 'light' | 'dark';
+};
+
+type TDroppableContainerProps = {
+  isDragging: boolean;
+};
 
 const DroppableContainer = styled.div`
   width: 100%;
   position: relative;
   max-width: ${vars.constraint10};
-  cursor: ${(props) => (props.isDragging ? 'grabbing' : 'auto')};
+  cursor: ${(props: TDroppableContainerProps) =>
```

Comment:
Perhaps instead of this we might pass the type like that:
`const DroppableContainer = styled.div<TDroppableContainerProps>`
but it's up to you

Reply:
Thanks for the suggestion, Kacper.
I'm not sure if we have a policy on this but I would argue that sometimes generics are not that readable to me. I mean, when you see a generic value applied you don't immediately know what that type is used for so you need to search for its purpose.

I know there can be done both ways and maybe it makes more sense if you use props several times in a component.
Anyway, I would like to know @Rhotimee and @ddouglasz opinion about this so we try to standarize.

Reply:
I think it can be done both ways and it might be more handy to pass as generics if props were used in multiple places in a component.
I see the generics pattern in other parts of the codebase. Maybe we can just do same for uniformity since `TDroppableContainerProps` already seems defined, I don't think in this case, it would be difficult to read the type. Either ways, I see no harm.

Reply:
Either is fine with me.  However, I see that we use the pattern @kark suggested more in the codebase though.

Reply:
So we have a consensus.
I'll update to use the generic pattern.

Thank you all 👍

Reply:
Updated in this commit 4d6e6f47f43f5fa062f50edc499ca50fd205e37e